### ISSUE TITLE
Creates an invalid setter for optional keys

### ISSUE THREAD

**Issue by tomharvey** [tbuehlmann/dry-struct-setters#2]
I'm using this library in my own library and a user has found an error in how it creates setters when used with a feature found in dry-struct > v1.0.

When declaring a struct with an [optional key](https://dry-rb.org/gems/dry-types/1.0/hash-schemas/#optional-keys) dry-struct-setter will create an unexpected and invalid setter.
```
class Foo
    attribute? :optional_key
end

foo.optional_key = 'new-value'
undefined method `optional_key=' for #<#<Class:0x00007fe179a12ea0>:0x00007fe179a08180>
       Did you mean?  optional_key?=
```

I would expect the setter to be `foo.optional=`, not `foo.optional?=`

And, `foo.optional? = 'new-value'` is somewhat invalid ruby.

See the original issue in my library here https://github.com/tomharvey/pwinty3-rb/issues/14

**Comment by tomharvey** [comment]
See the PR https://github.com/tbuehlmann/dry-struct-setters/issues/2 for more description and a potential solution.

**Comment by tbuehlmann** [comment]
Great, thanks! Just reviewed the PR and added some suggestions.

**Comment by tbuehlmann** [comment]
Solved in 3e311bc94fa8e2879c2c19967f8b4072be6a2c76.
